modules/azure_subscription/main.tf

```
@@ -0,0 +1,152 @@
+        "Microsoft.Network/firewallPolicies/ruleCollectionGroups/delete",
+        "Microsoft.Network/firewallPolicies/ruleGroups/write",
+        "Microsoft.Network/firewallPolicies/ruleGroups/delete",
+        "Microsoft.Network/ipGroups/write",
+        "Microsoft.Network/ipGroups/validate/action",
+        "Microsoft.Network/ipGroups/updateReferences/action",
+        "Microsoft.Network/ipGroups/join/action",
+        "Microsoft.Network/ipGroups/delete"
+      ] : []
+    )
+  }
+
+  assignable_scopes = ["${local.subscription_scope}/${var.subscription_id}"]
+  scope             = "${local.subscription_scope}/${var.subscription_id}"
+}
+
+# Assigning Role for Firewall
+resource "azurerm_role_assignment" "illumio_fw_assignment" {
+  principal_id       = azuread_service_principal.illumio_sp.object_id
+  role_definition_id = azurerm_role_definition.illumio_fw_role.role_definition_resource_id
+  scope              = "${local.subscription_scope}/${var.subscription_id}"
+}
+
+# Role Definitions for NSG
+resource "azurerm_role_definition" "illumio_nsg_role" {
+  name        = var.nsg_role_name
+  description = "Illumio Network Security Administrator role"
+
+  permissions {
+    actions = concat(
+      [
+        "Microsoft.Network/networkInterfaces/effectiveNetworkSecurityGroups/action",
+        "Microsoft.Network/networkSecurityGroups/read",
+        "Microsoft.Network/networkSecurityGroups/defaultSecurityRules/read",
+        "Microsoft.Network/networksecuritygroups/providers/Microsoft.Insights/diagnosticSettings/read",
+        "Microsoft.Network/networksecuritygroups/providers/Microsoft.Insights/logDefinitions/read",
+        "Microsoft.Network/networkWatchers/securityGroupView/action"
+      ],
+      var.mode == "ReadWrite" ? [
+        "Microsoft.Network/networkSecurityGroups/write",
+        "Microsoft.Network/networkSecurityGroups/delete",
+        "Microsoft.Network/networkSecurityGroups/securityRules/write",
+        "Microsoft.Network/networkSecurityGroups/securityRules/delete",
+        "Microsoft.Network/networksecuritygroups/providers/Microsoft.Insights/diagnosticSettings/write"
+      ] : []
+    )
+  }
+
+  assignable_scopes = ["${local.subscription_scope}/${var.subscription_id}"]
+  scope             = "${local.subscription_scope}/${var.subscription_id}"
+}
+
+# Assigning Role for NSG
+resource "azurerm_role_assignment" "illumio_nsg_assignment" {
+  principal_id       = azuread_service_principal.illumio_sp.object_id
+  role_definition_id = azurerm_role_definition.illumio_nsg_role.role_definition_resource_id
+  scope              = "${local.subscription_scope}/${var.subscription_id}"
+}
+
+
+resource "illumio-cloudsecure_azure_subscription" "subscription" {
+  client_id       = azuread_application.illumio_app.client_id
+  client_secret   = base64encode(azuread_application_password.illumio_secret.value)
+  name            = var.name
+  subscription_id = var.subscription_id
+  tenant_id       = var.tenant_id
```

Get those from the current subscription, to be consistent with the `aws_account` module.
```suggestion
  subscription_id = data.azurerm_subscription.current.subscription_id
  tenant_id       = data.azurerm_subscription.current.tenant_id
```